chromadb/auth/token/__init__.py

```diff
@@ -0,0 +1,183 @@
+import logging
+from enum import Enum
+from typing import Tuple, Any, cast, Dict, TypeVar
+
+from overrides import override
+from pydantic import SecretStr
+
+from chromadb.auth import (
+    ServerAuthProvider,
+    ClientAuthProvider,
+    ServerAuthenticationRequest,
+    ServerAuthCredentialsProvider,
+    AuthInfoType,
+    ClientAuthCredentialsProvider,
+    ClientAuthResponse,
+    SecretStrAbstractCredentials,
+    AbstractCredentials,
+)
+from chromadb.auth.registry import register_provider, resolve_provider
+from chromadb.config import System
+from chromadb.utils import get_class
+
+T = TypeVar("T")
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["TokenAuthServerProvider", "TokenAuthClientProvider"]
+
+_token_transport_headers = ["Authorization", "X-Chroma-Token"]
+
+
+# add new enum for token transport headerts
+class TokenTransportHeader(Enum):
+    AUTHORIZATION = "Authorization"
+    X_CHROMA_TOKEN = "X-Chroma-Token"
```

Comment:
Just to confirm my understanding, is the reason to support both in the case of you using Bearer Auth somewhere else in your system?

Reply:
Yeah. It could be that the user has an intermediate auth proxy which can use the bearer then they should use the X header.